### PR TITLE
detect openrisc core, support orbis32, orbis64, orfpx32, orfpx64, orvdx64

### PR DESCRIPTION
**Pull request by YuzukiTsuru**

### CODE REVIEW

File: ruapu.h [hunk 1]
```diff
@@ -371,6 +378,15 @@ void ruapu_init()
...
     }
 #else
+#if defined __openrisc__
+	uint32_t value;
+    uint16_t addr = U(0x0000);
+	asm volatile ("l.mfspr %0, r0, %1" : "=r" (value) : "K" (addr));
+    for (size_t i = 0; i < sizeof(g_ruapu_isa_map) / sizeof(g_ruapu_isa_map[0]); i++)
+    {
+        g_ruapu_isa_map[0].capable = ((value) >> (5 + i)) & 0x1;
+    }
```
Comment (nihui) [inline ruapu.h:388]: space instead of tab

File: ruapu.h [hunk 2]
```diff
@@ -371,6 +378,15 @@ void ruapu_init()
         g_ruapu_isa_map[i].capable = ruapu_detect_isa(g_ruapu_isa_map[i].inst);
     }
 #else
+#if defined __openrisc__
```
Comment (nihui) [inline ruapu.h:381]: wrap these code into a static function, call it from here

Comment (YuzukiTsuru) [reply ruapu.h:394]: done

**Comment by nihui** [comment]
Thanks for your contribution !

**Pull request merged**